string color1
string color2
---
string[] colors
float32[] probabilities